src/model/consent/consent.ts

```diff
@@ -83,20 +83,51 @@ export class ConsentDB {
   }
 
   // Update Consent
-  // No validation against Null or illegal updates in models
   public async update (consent: Consent): Promise<number> {
     // Returns number of updated rows
-    const updateCount: number = await this
-      .Db<Consent>('Consent')
-      .where({ id: consent.id })
-      .update(consent)
-
-    // Ensure that the caller knows that the resource does not exist
-    if (updateCount === 0) {
-      throw new NotFoundError('Consent', consent.id)
-    }
-
-    return updateCount
+    // Transaction to make the update atomic
+    return this.Db.transaction(async (trx): Promise<number> => {
+      // Transaction is rolled back automatically if there is
+      // an error and the returned promise is rejected
+      const consents: Consent[] = await trx<Consent>('Consent')
+        .select('*')
+        .where({ id: consent.id })
+        .limit(1)
```

Comment:
This isn't necessary, right?

Reply:
You suggested having a limit for `select` with `Consent` which is a unique resource. That is why we also do the same in `retrieve`.